### ISSUE TITLE
GCP: GKE with spot instances in nodepool is detected as on-demand

### ISSUE THREAD

**Issue by syphernl** [infracost/infracost#2359]
While setting up a new project on GCP I noticed that Infracost detects spot instances in a GKE Nodepool as "on-demand" rather than spot nodes.

Example Terraform code:

```tf
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google"
  version = "25.0.0"

  project_id                      = "test"
  name                            = "test-cluster"
  regional                        = false
  region                          = "europe-west4"
  zones                           = ["europe-west4-a"]
  network                         = "my-test-network"
  subnetwork                      = "my-subnetwork"
  ip_range_pods                   = "my-iprange"
  ip_range_services               = "my-services"
  release_channel                 = "REGULAR"
  remove_default_node_pool        = true

  node_pools = [
    {
      name               = "spot-nodes"
      spot               = true
      initial_node_count = 0
      max_count          = 3
      disk_size_gb       = 25
      enable_secure_boot = true
      disk_type          = "pd-ssd"
      machine_type       = "n1-standard-1"
    }
  ]
}
```

This produces:

```bash
Evaluating Terraform directory at .
  ✔ Downloading Terraform modules 
  ✔ Evaluating Terraform directory 
  ✔ Retrieving cloud prices to calculate costs 

Project: .

 Name                                                       Monthly Qty  Unit   Monthly Cost (EUR) 
                                                                                                   
 module.gke.google_container_cluster.primary                                                       
 ├─ Cluster management fee                                          730  hours              €68.98 
                                                                                                   
 module.gke.google_container_node_pool.pools["spot-nodes"]                                         
 ├─ Instance usage (Linux/UNIX, on-demand, n1-standard-1)           730  hours              €25.25 
 └─ SSD provisioned storage (pd-ssd)                                 25  GB                  €4.42 
                                                                                                   
 OVERALL TOTAL (EUR)                                                                        €98.65 
──────────────────────────────────
7 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 5 were free, rerun with --show-skipped to see details
```

According to the Google Pricing Calculator running a `n1-standard-1` as spot for 730 hours should be `EUR 6,18`  and not `EUR 25,25`

**Comment by aliscott** [comment]
Thanks for reporting @syphernl. Do you have example Terraform code we can use to reproduce this?

**Comment by syphernl** [comment]
> Thanks for reporting @syphernl. Do you have example Terraform code we can use to reproduce this?

Certainly! I have updated my initial message.

**Comment by aliscott** [comment]
This is fixed in https://github.com/infracost/infracost/pull/2447 and will go out in the next release.

**Comment by hugorut** [comment]
the fix for this is now live in the [v0.10.22](https://github.com/infracost/infracost/releases/tag/v0.10.22) release.
